emsdk.py

```diff
@@ -2421,20 +2418,32 @@ def set_active_tools(tools_to_activate, permanently_activate, system):
   # calling shell environment.  On other platform `source emsdk_env.sh` is
   # required.
   if WINDOWS:
-    env_string = construct_env(tools_to_activate)
-    write_set_env_script(env_string)
+    if not permanently_activate:
+      env_string = construct_env(tools_to_activate)
+      write_set_env_script(env_string)
+    else:
+      env_vars_to_add = get_env_vars_to_add(tools_to_activate)
 
-  # Apply environment variables to global all users section.
-  if WINDOWS and permanently_activate:
-    # Individual env. vars
-    for tool in tools_to_activate:
-      tool.win_activate_env_vars(system=system)
+      if len(env_vars_to_add) > 0:
+        changed = False
 
-    # PATH variable
-    newpath, added_items = adjusted_path(tools_to_activate, system=system)
-    # Are there any actual changes?
-    if newpath != os.environ['PATH']:
-      win_set_environment_variable('PATH', newpath, system=system)
+        for key, value in env_vars_to_add:
+          if win_set_environment_variable(key, value, system=system):
+            if not changed:
+              changed = True
+              print('Setting global environment variables:')
+
+            print(key + ' = ' + value)
+
+        if not changed:
+          print('Global environment variables up to date')
+        else:
+          try:
+            import win32gui
+            import win32con
+            win32gui.SendMessageTimeout(win32con.HWND_BROADCAST, win32con.WM_SETTINGCHANGE, 0, 'Environment', win32con.SMTO_BLOCK, 100)
+          except Exception as e:
+            errlog('SendMessageTimeout failed with error: ' + str(e))
```

Comment:
Might be nice to keep this entire block as a seperate function.   Then you could avoid a lot of the nesting by having early returns after `if not changed:` and after `if env_vars_to_add:`?